proof_ipa.go

```diff
@@ -138,7 +138,7 @@ func SerializeProof(proof *Proof) ([]byte, []KeyValuePair, error) {
 
```

Comment:
Remember that for proofs, we still use compressed serialization.

Reply:
proofs and commitments computation alike. With this approach, I get a different root commitment, so something must impact commitment calculation, when it shouldn't.

Reply:
All tests, proof gen, and verification still pass.
One of the tests is checking compatibility with Rust and is still passing.

This sounds more like a `go-ethereum` call to the changed semantics `Bytes()` as we chatted; we can look at it.